crates/worker/src/partition_processor_manager.rs

```diff
@@ -557,59 +608,67 @@ impl PartitionProcessorManager {
+                    gauge!(PARTITION_LAST_APPLIED_LOG_LSN,
                         PARTITION_LABEL => partition_id.to_string())
-                        .set(last_applied_log_lsn.as_u64() as f64);
-                    }
+                    .set(last_applied_log_lsn.as_u64() as f64);
+                }
 
-                    if let Some(last_persisted_log_lsn) = status.last_persisted_log_lsn {
-                        gauge!(PARTITION_LAST_PERSISTED_LOG_LSN,
+                if let Some(last_persisted_log_lsn) = status.last_persisted_log_lsn {
+                    gauge!(PARTITION_LAST_PERSISTED_LOG_LSN,
                         PARTITION_LABEL => partition_id.to_string())
-                        .set(last_persisted_log_lsn.as_u64() as f64);
-                    }
+                    .set(last_persisted_log_lsn.as_u64() as f64);
+                }
 
-                    if let Some(last_record_applied_at) = status.last_record_applied_at {
-                        gauge!(PARTITION_TIME_SINCE_LAST_RECORD,
+                if let Some(last_record_applied_at) = status.last_record_applied_at {
+                    gauge!(PARTITION_TIME_SINCE_LAST_RECORD,
                         PARTITION_LABEL => partition_id.to_string())
-                        .set(last_record_applied_at.elapsed());
-                    }
+                    .set(last_record_applied_at.elapsed());
+                }
 
-                    // it is a bit unfortunate that we share PartitionProcessorStatus between the
-                    // PP and the PPManager :-(. Maybe at some point we want to split the struct for it.
-                    status.last_persisted_log_lsn = persisted_lsns
-                        .as_ref()
-                        .and_then(|lsns| lsns.get(partition_id).cloned());
+                // it is a bit unfortunate that we share PartitionProcessorStatus between the
+                // PP and the PPManager :-(. Maybe at some point we want to split the struct for it.
+                status.last_persisted_log_lsn = persisted_lsns
+                    .as_ref()
+                    .and_then(|lsns| lsns.get(partition_id).cloned());
 
-                    status.last_archived_log_lsn = self.archived_lsns.get(partition_id).cloned();
+                status.last_archived_log_lsn = self.archived_lsns.get(partition_id).cloned();
 
-                    Some((*partition_id, status))
-                } else {
-                    None
+                let target_tail_lsn = self.target_tail_lsns.get(partition_id).cloned();
+                if target_tail_lsn > status.target_tail_lsn {
+                    status.target_tail_lsn = target_tail_lsn;
+                }
+
+                if let Some((target_lsn, applied_lsn)) =
+                    status.target_tail_lsn.zip(status.last_applied_log_lsn)
+                {
+                    gauge!(PARTITION_LAST_APPLIED_LSN_LAG, PARTITION_LABEL => partition_id.to_string())
```

Comment:
Should the metrics also be updated when there is no external component asking for `get_state`?